hera_pspec/pspecbeam.py

```diff
@@ -0,0 +1,291 @@
+        a beam area (in str). Uses analytic formula that the answer
+        is 2 * pi * FWHM**2 / 8 ln 2.
+
+        Trivially this returns an array (i.e., a function of frequency),
+        but the results are frequency independent.
+
+        Parameters
+        ----------
+        stokes: str, optional
+                Which Stokes parameter's beam to compute the scalar for.
+                'I', 'Q', 'U', 'V', although currently only 'I' is implemented
+                Default: 'I'
+        Returns
+        -------
+        primary beam area: float, array-like
+        """
+        if stokes != 'I':
+            raise NotImplementedError()
+        else:
+            return np.ones_like(self.beam_freqs) * 2. * np.pi * self.FWHM**2 / (8. * np.log(2.))
+
+    def power_beam_sq_int(self, stokes='I'):
+        """
+        Computes the integral of the beam**2 over solid angle to give
+        a beam area (in str). Uses analytic formula that the answer
+        is pi * FWHM**2 / 8 ln 2.
+
+        Trivially this returns an array (i.e., a function of frequency),
+        but the results are frequency independent.
+
+        Parameters
+        ----------
+        stokes: str, optional
+                Which Stokes parameter's beam to compute the scalar for.
+                'I', 'Q', 'U', 'V', although currently only 'I' is implemented
+                Default: 'I'
+        Returns
+        -------
+        primary beam area: float, array-like
+        """
+        if stokes != 'I':
+            raise NotImplementedError()
+        else:
+            return np.ones_like(self.beam_freqs) * np.pi * self.FWHM**2 / (8. * np.log(2.))
+
+    def compute_pspec_scalar(self, lower_freq, upper_freq, num_freqs, stokes='I', taper='none', little_h=True, num_steps=10000):
+        """
+        Computes the scalar function to convert a power spectrum estimate
+        in "telescope units" to cosmological units
+
+        See arxiv:1304.4991 and HERA memo #27 for details.
+
+        Currently this is only for Stokes I.
+
+        Parameters
+        ----------
+        lower_freq: float
+                Bottom edge of frequency band over which power spectrum is
+                being estimated. Assumed to be in Hz.
+
+        upper_freq: float
+                Top edge of frequency band over which power spectrum is
+                being estimated. Assumed to be in Hz.
+
+        num_freqs: int
+                Number of frequency channels in the data.
+
+        stokes: str, optional
+                Which Stokes parameter's beam to compute the scalar for.
+                'I', 'Q', 'U', 'V', although currently only 'I' is implemented
+                Default: 'I'
+
+        taper : str, optional
+                Whether a tapering function (e.g. Blackman-Harris) is being
+                used in the power spectrum estimation.
+                Default: none
+
+        little_h : boolean, optional
```

Comment:
Can you mention where the value of `h` is taken from, please?

Reply:
Made the conversions object something that is carried around by pspecbeam, as we've discussed.